generator/mavgen_javascript.py

```diff
@@ -234,35 +245,44 @@ def field_descriptions(fields):
         outf.write("""
 
     this.format = '%s';
-    this.id = %s.MAVLINK_MSG_ID_%s;
```

Comment:
This will break people's code pretty horribly.

Likewise for _name.

Probably better to leave this in place, but with a comment explaining why people should use `_id` and `_name`.

I can't see a technical way to deprecate use of these attributes with name collisions.

Reply:
Yes, i'm totally worried about the attribute collisions, as we have a bunch of them without my changes...  the list below are all places where the old javascript code collides between xml field names and internal parser goodness, which is very bad, and should absolutely not happen.   right now, i've had to break backward compatibility in this commit because
(example/s) 
 - with a message of type BATTERY_STATUS -> msg.id  refers to field 'Battery ID'  not MsgID which was also msg.id before this pr
 - with a messages of type PLAY_TUNE_V2  ->  msg.format refers to bitmask of supported tune formats enum="TUNE_FORMAT" , not the parser's FMT string, which was also msg.format before this pr
 and this happens all over the place.

The other alternative/s dont look any better, for example if we moved all the xml attributes to something like msg.field.id and msg.field.format and msg.field.name then that still breaks backward compatibility because there's gonna be a buncg of code that uses (eg) msg.name and msg.id  , and there's not way to magically fix this.   

The solution ive used here follows the python parser's convention/s as closely as reasonably possible... python tends to prefix internal attributes with an _  ( like _msgbuf ), so thats also one of the reasons i chose this way

current problematic field/s i've resolved in this pr:
name => _name
id => _id
format => _format
payload => _payload 
header => _header
data => _data

I have thought about making a dynamic gettter/setter for the attribute/s of a packet , so that it would maintain compatibility for as many packets as possible by looking for an attribute *first* in the xml attributes  and then in the header/s, essentially meaning that we've encoded a 'preference' that (example) "msg.id"  should mean msg._id most of the time but in packet/s that have an 'id' field in their xml, it would instead mean msg.fields.id  , but although that's a pretty smart backward/s compatability shim that could work for most things most of the time, it's utterly non-compliant. 

Sooo... the only answer i have is to make a one-off breaking change to the parser, that actually fixes it properly, aligns it more with the way python does it, and importantly, gets the parser passing all its tests, which it never had before.